internal/server/server.go

```diff
@@ -18,8 +24,26 @@ import (
 //go:embed templates
 var embeddedFS embed.FS
 
+type ExternalFuncs struct {
```

Comment:
This is the right approach with mocking. My recommendation is to switch to  using an `interface` instead of a struct is more idiomatic golang and will also clean up the code. 

e.g.

This interface represents the three functions that appear in the oauth2.Config object.  This way you can define within the tests a mock object that implements these functions.

```
type OIDCConfigProvider interface {
	AuthCodeURL(state string, opts ...oauth2.AuthCodeOption) string
	Exchange(ctx context.Context, code string, opts ...oauth2.AuthCodeOption) (*oauth2.Token, error)
	Client(ctx context.Context, t *oauth2.Token) *http.Client
}
```

Sample test setup
```
type MockOIDCServer struct {
}

func (m *MockOIDCServer) AuthCodeURL(state string, opts ...oauth2.AuthCodeOption) string {
	return "test"
}

....
server.oauth2Provider = mock
```

Reply:
Fixed, thanks for all the help